src/components/form/TextFieldBase.vue

```diff
@@ -0,0 +1,171 @@
+<template>
+  <FieldBase
+    v-bind="$attrs"
+    class="text-field"
+    @focusin.native="onFieldFocusIn"
+    @focusout.native="onFieldFocusOut"
+  >
+    <slot />
+    <template #input-right>
+      <slot
+        v-if="$slots['input-right']"
+        name="input-right"
+      />
+      <HdIcon
+        v-else-if="isValidationCheckVisible"
+        :src="checkIcon"
+        width="24"
+        height="24"
+        class="text-field__status-icon text-field__status-icon--valid"
+        @click.native="$emit('status-click')"
+      />
+      <button
+        v-else-if="isClearButtonVisible"
+        class="text-field__clear-button"
+        @click="$emit('clear-click')"
```

Comment:
Maybe just `$emit('clear')` would be enough in this case:

`<TextFieldBase @clear="" />` is already good without the `-click` postfix

Reply:
The thing is that the event that happens is the click, not the clearing, because the clearing happens on the parent level.
We are just reporting that the "clear" button has been clicked (the clearing didn't happen yet). We can also think of a better name, but I'm afraid it will still have the word "click" 😕

Reply:
Ook, not a problem 😄